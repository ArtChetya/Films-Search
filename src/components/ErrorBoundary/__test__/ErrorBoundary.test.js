import { shallow } from 'enzyme/build';
import React from 'react';
import { ErrorBoundary } from '..';

describe('ErrorBoundary component', () => {
    it('should be rendered correctly', () => {
        const component = shallow(
            <ErrorBoundary>
                <div>Test</div>
            </ErrorBoundary>
        );
        expect(component).toMatchSnapshot();
    });

    it('should be rendered correctly when child have error', () => {
        const ComponentWithError = () => {
            throw new Error('Errored!');
        };

        const component = shallow(
            <ErrorBoundary>
                <ComponentWithError />
            </ErrorBoundary>
        );
        expect(component).toMatchSnapshot();
    });
});
