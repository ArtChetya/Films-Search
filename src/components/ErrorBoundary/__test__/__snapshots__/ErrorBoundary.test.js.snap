// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorBoundary component should be rendered correctly 1`] = `
<div>
  Test
</div>
`;
